opta/cloud/__init__.py

```diff
@@ -0,0 +1,54 @@
+# new-module-api
+
+from opta.core.terraform2 import StateStore, StoreConfig
+from opta.layer2 import Layer
+
+from .aws import AWSProvider, S3Store
+from .local import LocalStore
+from .provider import CloudProvider
+
+
+def cloud_provider_for_layer(layer: Layer) -> CloudProvider:
+    """
+    Returns a new CloudProvider based on the layer configuration.
+    """
+    cloud_id = layer.providers.cloud_id
+    if not cloud_id:
+        raise ValueError("No cloud provider configured")
+
+    if cloud_id == "aws":
+        config = layer.providers.aws
+        assert config is not None
+
+        return AWSProvider(config)
+
+    raise ValueError(f"Unknown cloud provider {cloud_id}")
```

Comment:
would that make sense to move this one to cloud_client?
https://github.com/run-x/opta/blob/main/opta%2Fcore%2Fcloud_client.py
since it looks like this is something that needs to be implemented for each cloud

also we recently added https://github.com/run-x/opta/blob/3960cd526ce82299dfeab42d366d7b2da48cbd7d/opta/layer.py#L176-L188